R/mainTests.r

```diff
@@ -1153,6 +1160,454 @@ country_code <- function(data) {
   return(output)
 }
 
+
+# school_laestab_duplicates  ----------------------------------------
+# check that there is a 1:1 relationship between school laestab codes and names
+school_laestab_duplicates <- function(data) {
+  if (!"school_name" %in% names(data)) {
+    output <- list(
+      "message" = "School-level data is not present in this data file.",
+      "result" = "IGNORE"
+    )
+  } else {
+    multi_count_code <- data %>%
+      select("school_laestab", "school_name") %>%
+      distinct() %>%
+      add_count(school_name, name = "school_name_n") %>%
+      filter(school_name_n > 1) %>%
+      mutate(combo = paste0(school_name, " - ", school_name_n, " different laestabs")) %>%
+      select(combo) %>%
+      distinct() %>%
+      pull()
+
+    multi_count_name <- data %>%
+      select("school_laestab", "school_name") %>%
+      distinct() %>%
+      add_count(school_laestab, name = "school_code_n") %>%
+      filter(school_code_n > 1) %>%
+      mutate(combo = paste(school_laestab, " - ", school_code_n, " different school names")) %>%
+      filter(school_code_n > 1) %>%
+      select(combo) %>%
+      distinct() %>%
+      pull()
+
+
+    if (length(multi_count_code) == 0 & length(multi_count_name) == 0) {
+      output <- list(
+        "message" = "every school name has one laestab, and vice versa.",
+        "result" = "PASS"
+      )
+    } else {
+      if (length(multi_count_code) == 1 & length(multi_count_name) == 0) {
+        output <- list(
+          "message" = paste0("The following school_name has multiple laestabs '", paste0(multi_count_code), "'. 
+                             <br> - Every school name should have only one laestab and vice versa."),
+          "result" = "FAIL"
+        )
+      } else {
+        if (length(multi_count_code) == 1 & length(multi_count_name) == 1) {
+          output <- list(
+            "message" = paste0("The following school_name has multiple laestabs '", paste0(multi_count_code), "', 
```

Comment:
I think these messages where you have words inside the ' ' we could drop the quote marks? They're useful to show specific variable values, but once we're adding text in there too I think they get a little confusing and clutter it a bit. Could add them in higher up when making the combo column, or drop them entirely.

Same then goes across the rest of the similar error messages

Reply:
Yes dropped these now - agree much neater this way